# Synthetic Config

# Initial standard deviations (attitude, velocity, position)
extended_pose_std: [1.0e-4, 1.0e-4, 1.0e-9, 1.0e-1, 1.0e-1, 1.0e-1, 1.0e-9, 1.0e-9, 1.0e-9]
bias_std: [1.0e-2, 1.0e-2, 1.0e-2, 1.0e-2, 1.0e-2, 1.0e-2]
extrinsics_std: [1.0e-3, 1.0e-3, 1.0e-3, 1.0e-3, 1.0e-3, 1.0e-3]
# intrinsics_std: [10, 10, 10, 10]

# IMU noise statistics
accelerometer_noise_density: 0.5e-2  
accelerometer_random_walk:   1.0e-3
gyroscope_noise_density: 1.745e-4
gyroscope_random_walk:   1.0e-04

# Camera calibration
distortion_coeffs: [0.0, 0.0, 0.0, 0.0]
distortion_model: radtan
resolution: [640, 480]
intrinsics: [270.0, 270.0, 320.0, 240.0]
T_imu_cam:
 - [0.0, 0.0, 1.0, 0.2]
 - [-1.0, 0.0, 0.0, 0.2]
 - [0.0, -1.0, 0.0, 0.05]
 - [0.0, 0.0, 0.0, 1.0]

# Initializer
static_initializer_imu_window: 0.0
static_initializer_disparity_window: 0.0
static_initializer_acc_threshold: -1.0
static_initializer_disparity_threshold: -1.0

# Propagator
state_transition_order: 1
imu_buffer_max_size: 1000

# Updater
refine_traingulation: true
feature_min_depth: 0.1
feature_max_depth: 50
feature_refinement_max_iterations: 20
feature_refinement_tollerance: 1e-9
measurement_projection_method: unit_plane
feature_representation: anchored_inverse_depth
pixel_standerd_deviation: 0.5
curveture_correction: false
min_angle_deg: 0.0

# State options
enable_camera_extrinsic_calibration: true
enable_camera_intrinsic_calibration: false
gravity: 9.81
num_clones: 11
num_persistent_features: 0

# Tracker
equalization_method: none
optical_flow_pyramid_levels: 3
feature_detector: fast
grid_x_size: 1
grid_y_size: 1
min_feature_pixel_distance: 5
min_features: 100
max_features: 100
fast_threshold: 5
shi_tomasi_quality_level: 0.01

# Track Manager
max_track_length: 250

# Logger level [0: Full, 1: INFO, 2: WARN, 3: ERR, 4: INACTIVE]
logger_level: 4
